### PR TITLE
Add support for multi-core thread_id in filebeat postgresql module

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
This PR is for https://github.com/elastic/beats/issues/9156
closes https://github.com/elastic/beats/issues/9156

### CODE REVIEW

**Comment by kaiyan-sheng** [comment]
`core_id` is probably a bad name here. I am totally open to a better name!

File: filebeat/module/postgresql/log/test/postgresql-9.6-debian-with-slowlog.log-expected.json [hunk 1]
```diff
@@ -6,8 +6,6 @@
         "input.type": "log", 
         "log.offset": 0, 
         "message": "2017-07-31 13:36:42.585 CEST [4974] LOG:  database system was shut down at 2017-06-17 16:58:04 CEST", 
-        "postgresql.log.level": "LOG",
```
Comment (ruflin) [inline filebeat/module/postgresql/log/test/postgresql-9.6-debian-with-slowlog.log-expected.json]: Why are these values missing now?

**Comment by sayden** [comment]
I have seen that the input log lines can be very tricky so don't forget that you can try with more than one Grok patterns too.

**Comment by kaiyan-sheng** [comment]
@ruflin yes for the changelog. Now I have this pr number, I can add it into the changelog :-)

File: filebeat/module/postgresql/log/test/postgresql-9.6-debian-with-slowlog.log-expected.json (hunk 1 above)
Comment (kaiyan-sheng) [reply filebeat/module/postgresql/log/test/postgresql-9.6-debian-with-slowlog.log-expected.json]: I broke it 👎

**Comment by kaiyan-sheng** [comment]
@sayden Good point! I just updated `filebeat/module/postgresql/log/test/postgresql-9.5-multi-core.log` with more log lines.

File: filebeat/module/postgresql/log/test/postgresql-ubuntu-9.5.log-expected.json [hunk 2]
```diff
@@ -0,0 +1,1092 @@
+[
```
Comment (jsoriano) [inline filebeat/module/postgresql/log/test/postgresql-ubuntu-9.5.log-expected.json]: No need to add this expected file, some longer test files are only to check that the content can be parsed, not to check the generated events.

File: filebeat/generator/fields/fields_test.go [hunk 3]
```diff
@@ -38,11 +38,11 @@ func TestFieldsGenerator(t *testing.T) {
...
+				"%{LOCALDATETIME:postgresql.log.timestamp} %{WORD:postgresql.log.timezone} \\[(%{NUMBER:postgresql.log.thread_id}|%{NUMBER:postgresql.log.thread_id}-%{BASE16FLOAT:postgresql.log.core_id})\\] %{USERNAME:postgresql.log.user}@%{HOSTNAME:postgresql.log.database} %{WORD:postgresql.log.level}:  ?%{GREEDYDATA:postgresql.log.message}",
+				"%{LOCALDATETIME:postgresql.log.timestamp} %{WORD:postgresql.log.timezone} \\[(%{NUMBER:postgresql.log.thread_id}|%{NUMBER:postgresql.log.thread_id}-%{BASE16FLOAT:postgresql.log.core_id})\\] \\[%{USERNAME:postgresql.log.user}\\]@\\[%{HOSTNAME:postgresql.log.database}\\] %{WORD:postgresql.log.level}:  ?%{GREEDYDATA:postgresql.log.message}",
+				"%{LOCALDATETIME:postgresql.log.timestamp} %{WORD:postgresql.log.timezone} \\[(%{NUMBER:postgresql.log.thread_id}|%{NUMBER:postgresql.log.thread_id}-%{BASE16FLOAT:postgresql.log.core_id})\\] %{WORD:postgresql.log.level}:  ?%{GREEDYDATA:postgresql.log.message}",
```
Comment (jsoriano) [inline filebeat/generator/fields/fields_test.go]: I think this is not related to the postgresql module, it is intended to test the fields generator, so these changes are probably not needed. But now that you have added them, should we add the `core_id` to the list of expected fields below?

File: filebeat/module/postgresql/log/ingest/pipeline.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
         "field": "message",
         "ignore_missing": true,
         "patterns": [
-            "^%{LOCALDATETIME:postgresql.log.timestamp} %{WORD:postgresql.log.timezone} \\[%{NUMBER:postgresql.log.thread_id}\\] ((\\[%{USERNAME:postgresql.log.user}\\]@\\[%{POSTGRESQL_DB_NAME:postgresql.log.database}\\]|%{USERNAME:postgresql.log.user}@%{POSTGRESQL_DB_NAME:postgresql.log.database}) )?%{WORD:postgresql.log.level}:  (duration: %{NUMBER:postgresql.log.duration} ms  statement: %{GREEDYDATA:postgresql.log.query}|%{GREEDYDATA:postgresql.log.message})"
+            "^%{LOCALDATETIME:postgresql.log.timestamp} %{WORD:postgresql.log.timezone} \\[(%{NUMBER:postgresql.log.thread_id}|%{NUMBER:postgresql.log.thread_id}-%{BASE16FLOAT:postgresql.log.core_id})\\] ((\\[%{USERNAME:postgresql.log.user}\\]@\\[%{POSTGRESQL_DB_NAME:postgresql.log.database}\\]|%{USERNAME:postgresql.log.user}@%{POSTGRESQL_DB_NAME:postgresql.log.database}) )?%{WORD:postgresql.log.level}:  (duration: %{NUMBER:postgresql.log.duration} ms  statement: %{GREEDYDATA:postgresql.log.query}|%{GREEDYDATA:postgresql.log.message})"
```
Comment (jsoriano) [inline filebeat/module/postgresql/log/ingest/pipeline.json]: Could we add the `core_id` as optional so `thread_id` appears only one in the expression? i.e: `\\[%{NUMBER:postgresql.log.thread_id}(-%{BASE16FLOAT:postgresql.log.core_id})?\\]`

Comment (kaiyan-sheng) [reply filebeat/module/postgresql/log/ingest/pipeline.json]: yes! good point! Thanks! I destroyed this pr somehow ... I will create a new PR instead...

**Comment by kaiyan-sheng** [comment]
Thanks for all the comments. I addressed them in https://github.com/elastic/beats/pull/9482 and closing this one... I somehow managed to destroy it again...
